orchagent/routeorch.cpp

```diff
@@ -1435,12 +1442,22 @@ bool RouteOrch::addRoute(RouteBulkContext& ctx, const NextHopGroupKey &nextHops)
             gRouteBulker.set_entry_attribute(&object_statuses.back(), &route_entry, &route_attr);
         }
 
-        route_attr.id = SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID;
-        route_attr.value.oid = next_hop_id;
+        if (newNhgIsFineGrained && prevNhgWasFineGrained)
+        {
+            /* Don't change route entry if the route is previously fine grained and new nhg is also fine grained. 
+             * We already modifed sai nhg objs as part of addModifyFgNhg to account for nhg change. */
+            object_statuses.emplace_back();
+            object_statuses.back() = SAI_STATUS_SUCCESS;
```

Comment:
You can combine 2 statements into one. Just `emplace_back(SAI_STATUS_SUCCESS)`

Reply:
Combined the statements